tests/backup/backup_kubevirt_test.go

```diff
@@ -1760,3 +1765,424 @@ var _ = Describe("{KubevirtInPlaceRestoreWithReplaceAndRetain}", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			restoreName = fmt.Sprintf("%s-%s", "vm-restore-replace", RandomString(6))
+			restoreNames = append(restoreNames, restoreName)
+			log.InfoD("Restoring the [%s] backup", backupName)
+			err = CreateRestoreWithReplacePolicyWithValidation(restoreName, backupName, make(map[string]string), DestinationClusterName, BackupOrgID, ctx, make(map[string]string), ReplacePolicyDelete, scheduledAppContexts)
+			log.FailOnError(err, "Validating restore [%s] with retain", restoreName)
+			time.Sleep(bootTime)
+			defer func() {
+				err := SetSourceKubeConfig()
+				log.FailOnError(err, "failed to switch context to source cluster")
+			}()
+			err = SetDestinationKubeConfig()
+			log.FailOnError(err, "failed to switch to context to destination cluster")
+			vms, err := GetAllVMsFromScheduledContexts(restoredAppContexts)
+			log.FailOnError(err, "Failed to get VMs from scheduled contexts")
+			for _, v := range vms {
+				name, err := GetVirtLauncherPodName(v)
+				log.FailOnError(err, "Fetching virt launcher pod name for VM - [%s]", v.Name)
+				log.InfoD("Virt launcher pod name for VM - [%s] in namespace [%s] is [%s]", v.Name, v.Namespace, name)
+				dash.VerifyFatal(name != virtLauncherPodMap[v.Namespace][v.Name], true, fmt.Sprintf("Verifying that VM [%s] in namespace [%s] restarted", v.Name, v.Namespace))
+				virtLauncherPodMap[v.Namespace][v.Name] = name
+			}
+		})
+
+		Step("Adding additional disks to the VMs and restarting it", func() {
+			log.InfoD("Adding additional disks to the VMs and restarting it")
+			var err error
+			// Creating a new slice of VMs because the VM UID would have changed after restart
+			vms, err := GetAllVMsFromScheduledContexts(scheduledAppContexts)
+			log.FailOnError(err, "Failed to get VMs from scheduled contexts")
+			for _, v := range vms {
+				pvcs, err = CreatePVCsForVM(v, numberOfAdditionalDisks, "kubevirt-sc-for-cirros-cd", "5Gi")
+				log.FailOnError(err, "Failed to create PVCs for VM [%s] in namespace [%s]", v.Name, v.Namespace)
+				for _, appCtx := range scheduledAppContexts {
+					specListInterfaces := make([]interface{}, len(pvcs))
+					for i, pvc := range pvcs {
+						// Converting each element to interface
+						specListInterfaces[i] = pvc
+					}
+					appCtx.App.SpecList = append(appCtx.App.SpecList, specListInterfaces...)
+				}
+				err := AddPVCsToVirtualMachine(v, pvcs)
+				log.FailOnError(err, "Failed to add PVCs to VM [%s] in namespace [%s]", v.Name, v.Namespace)
+				err = RestartKubevirtVM(v.Name, v.Namespace, true)
+				log.FailOnError(err, "Failed to restart VM [%s] in namespace [%s]", v.Name, v.Namespace)
+				// Waiting for VMs to restart and complete boot process otherwise the disk count will be incorrect
+				log.Infof("Waiting for VM [%s] in namespace [%s] to boot. Sleeping for %d minutes...", v.Name, v.Namespace, bootTime)
+				time.Sleep(bootTime)
```

Comment:
How about? The current logic appears to work because the Inst().AppList contains only a single, hardcoded app.

```go
for _, appCtx := range scheduledAppContexts {
	vms, err := GetAllVMsFromScheduledContexts([]AppContext{appCtx})
	log.FailOnError(err, "Failed to get VMs from scheduled contexts")
	for _, v := range vms {
		pvcs, err := CreatePVCsForVM(v, numberOfAdditionalDisks, "kubevirt-sc-for-cirros-cd", "5Gi")
		log.FailOnError(err, "Failed to create PVCs for VM [%s] in namespace [%s]", v.Name, v.Namespace)
		specListInterfaces := make([]interface{}, len(pvcs))
		for i, pvc := range pvcs {
			// Converting each PVC to interface for appending to SpecList
			specListInterfaces[i] = pvc
		}
		appCtx.App.SpecList = append(appCtx.App.SpecList, specListInterfaces...)

		err = AddPVCsToVirtualMachine(v, pvcs)
		log.FailOnError(err, "Failed to add PVCs to VM [%s] in namespace [%s]", v.Name, v.Namespace)

		err = RestartKubevirtVM(v.Name, v.Namespace, true)
		log.FailOnError(err, "Failed to restart VM [%s] in namespace [%s]", v.Name, v.Namespace)
		
		// Perhaps moving it outside all loops is more efficient.
		log.Infof("Waiting for VM [%s] in namespace [%s] to boot. Sleeping for %d minutes...", v.Name, v.Namespace, bootTime)
		time.Sleep(bootTime)
	}
}
```

Reply:
How does it matter? All I need is the list of namespaces from the scheduled contexts. From those namespaces, I will get the VMs in it.

Reply:
Done